src/Validation/src/RuntimeValidatableTypeInfoResolver.cs

```diff
@@ -0,0 +1,390 @@
+            var typeValidationAttributes = type
+                .GetCustomAttributes<ValidationAttribute>()
+                .ToArray();
+
+            // Get all public instance properties
+            var properties = type.GetProperties(BindingFlags.Public | BindingFlags.Instance);
+            var validatableProperties = new List<RuntimeValidatablePropertyInfo>();
+
+            foreach (var property in properties)
+            {
+                // Skip properties without setters (read-only properties) for normal classes
+                if (!property.CanWrite && !IsRecordType(type))
+                {
+                    continue;
+                }
+
+                // Skip properties marked with [FromService] or [FromKeyedService] attributes
+                if (HasFromServiceAttributes(property.GetCustomAttributes()))
+                {
+                    continue;
+                }
+
+                // Get validation attributes for this property
+                var propertyValidationAttributes = property
+                    .GetCustomAttributes<ValidationAttribute>()
+                    .ToArray();
+
+                // For record types, also check constructor parameters for validation attributes
+                if (propertyValidationAttributes.Length == 0 && IsRecordType(type))
+                {
+                    var constructorValidationAttributes = GetValidationAttributesFromConstructorParameter(type, property.Name);
+                    if (constructorValidationAttributes.Length > 0)
+                    {
+                        propertyValidationAttributes = constructorValidationAttributes;
+                    }
+                }
+
+                // Get display name
+                var displayName = GetDisplayNameForProperty(property);
+
+                // Determine if this property has a validatable type
+                // Use the simpler check that doesn't cause recursion
+                var hasValidatableType = IsClassForType(property.PropertyType);
+
+                // Create the property info if it has validation attributes or a validatable type
+                if (propertyValidationAttributes.Length > 0 || hasValidatableType)
+                {
+                    var propertyInfo = new RuntimeValidatablePropertyInfo(
+                        declaringType: type,
+                        propertyType: property.PropertyType,
+                        name: property.Name,
+                        displayName: displayName,
+                        validationAttributes: propertyValidationAttributes);
+
+                    validatableProperties.Add(propertyInfo);
+                }
+            }
+
+            // Check for polymorphic derived types (JsonDerivedType attributes)
+            var derivedTypes = GetDerivedTypes(type);
+            foreach (var derivedType in derivedTypes)
+            {
+                // Recursively ensure derived types are also cached
```

Comment:
What's the reason for this? Wouldn't they be added to the cache once they are used?

Also, I don't see how these are being added to the cache. The return value is ignored.